### PR TITLE
Feature: Pending Applications Table

### PR DESCRIPTION
**Pull request by kennedybryant**
## Description

[Pull Request Video](https://www.loom.com/share/c8a1b4f3142e48d08f5e7326b857f125)

As an admin, I can view collected data from user signups in order to approve mentor/mentee applications. A Pending Applications folder and file were created to store the Pending Application component. This component receives data from the backend pertaining to all of the mentor and mentee applications that have been submitted but not yet approved for admins to view. A table is rendered, displaying name, role, and date of application submission. The component is properly linked to the nav sidebar "Pending Applications". 

**Note:** Backend is communicating with the frontend locally. The backend repo branch feature/pending-applications is needed for this component to receive data. 

Fixes [#107](https://trello.com/c/gmkzogVm)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by Shane-Gray394 — APPROVED** [review]
Looks great to me. Not entirely sure why you're getting the linting error, but it would seem every FE PR has that same issue.

**Review by lanceturbes — APPROVED** [review]
Wonderful! You inserted the new component right where our previous engineers left their spot for it, and your code is easy to read and understand.

I've checked out the branch locally on my PC, and swapped over to the "pending applications" feature branch for the back-end, as per your video, and no errors are cropping -- this looks ready to me.

**Review by jimlemoine — APPROVED** [review]
This PR looks ready to merge. Clearly documents the motivation and what changed, has well-named commits, is appropriately scoped, and has super clean code consistent with the rest of the project.

File: src/components/pages/PendingApplications/PendingApplication.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  },
+];
+
+const PendingApplications = () => {
+  const [applications, setApplications] = useState([]);
+
+  useEffect(() => {
+    const getPendingApps = () => {
+      axiosWithAuth()
+        .get('http://localhost:8080/application')
```
Comment (paulstgermain) [inline src/components/pages/PendingApplications/PendingApplication.js:43]: The only thing I'm seeing here is that we should be making use of our environment variables - by hardcoding `localhost` into this `get` request you've ensured that this won't work in deployment! 😱

Highly recommend making a patch to fix this as soon as you're free to!

**Review by paulstgermain — CHANGES_REQUESTED** [review]
This is highly organized code, and a classical implementation of Ant Designs table - great work!

The only thing I can see wrong here is mentioned below - it's a small thing that any of us could do, but could cause big problems down the line. Let's fix this before we merge this PR!

File: src/components/pages/PendingApplications/PendingApplication.js (hunk 1 above)
Comment (kennedybryant) [reply src/components/pages/PendingApplications/PendingApplication.js]: Thank you for pointing this out! I've pushed a commit that should fix this issue.

**Comment by mannig1224** [comment]
Great job on creating the component! Functionality-wise seems to be doing the job.

File: src/components/pages/PendingApplications/PendingApplication.js (hunk 1 above)
Comment (mannig1224) [reply src/components/pages/PendingApplications/PendingApplication.js]: @kennedybryant were you able to get this resolved?

**Review by mannig1224 — APPROVED** [review]
Great job! I saw that you resolved Pauls request so I went ahead and put that as resolved.

**Review by johnfigs — APPROVED** [review]
Will the recommendation from Paul being resolved, this PR look ready to go. It documents that motivation and what changed. Very nice progression and documentaiton of commits.

**Review by paulstgermain — APPROVED** [review]

**Pull request merged**